presto-main/src/main/java/com/facebook/presto/sql/planner/PlanFragmenter.java

```diff
@@ -918,6 +926,11 @@ public FragmentProperties setDistribution(PartitioningHandle distribution, Metad
                 return this;
             }
 
+            if (isPreferLocalUnion(session) && metadata.isRefinedPartitioningOver(session, currentPartitioning, distribution)) {
```

Comment:
This is added to support the case 
`bucketed table with 32 bucket UNION all bucketed table with 16 bucket`, the PartitioningHandle is set to table with 16 buckets